### DISCUSSION TITLE
Is there an equivalent to casks' uninstall stanza for formulae?

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#593] [Tap maintenance and Homebrew development] [answered]
Just wondering.

**Comment by carlocab** [comment, marked as answer]
No, there isn't.

**Reply by cesarcoatl** [reply]
Thanks.

Although this begs the question.

What could be done if some cleanup had to be performed before uninstall?

Additionally. Is there a reference similar to casks' [Stanza order](https://github.com/Homebrew/homebrew-cask/tree/master/doc/cask_language_reference#stanza-order) for formulae?

The Formula Cookbook is not as thorough.

Thanks again.

**Reply by carlocab** [reply]
>What could be done if some cleanup had to be performed before uninstall?

Make sure this isn't needed at install time by installing only to a formula's prefix (and avoiding `etc`, IIRC). Not sure there's much else you can do other than this.

>Is there a reference similar to casks' Stanza order for formulae?

Not exactly. You can use `brew audit --strict`, and it'll tell you if things are out of place. Or, you can have a look at the audit code; the required order is defined somewhere there.

**Reply by Rylan12** [reply]
There are so many methods that can be called in formulae that I think a list wouldn't be that practical. Especially given that `brew style --fix` will fix it for you.

In case you're curious, though, you can see the order (ruby array form) [here](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/ast_constants.rb).

**Reply by cesarcoatl** [reply]
Thanks again, @Rylan12.
